commands/doit.go

```diff
@@ -93,7 +105,26 @@ func initConfig() {
 	}
 
 	viper.SetDefault("output", "text")
+}
+
+func findConfig() (string, error) {
+	if cfgFile != "" {
+		return cfgFile, nil
+	}
 
+	legacyConfigPath := filepath.Join(homeDir(), ".doctlcfg")
+	if _, err := os.Stat(legacyConfigPath); err == nil {
+		msg := fmt.Sprintf("Configuration detected at %q. Please move .doctlcfg to %s/doctlcfg.yaml",
+			legacyConfigPath, configHome())
+		warn(msg)
+	}
+
+	ch := configHome()
+	if err := os.MkdirAll(ch, 0755); err != nil {
+		return "", err
+	}
+
+	return filepath.Join(ch, "doctlcfg.yaml"), nil
```

Comment:
I think `config.yaml` would be a more conventional name, given that other users of the XDG locations call their config's `myapp/config.ext`